SpatialGDKEditorToolbar/Source/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -79,13 +76,81 @@ bool CheckClassNameListValidity(const TArray<UClass*>& Classes)
 }
 }// ::
 
-void GenerateSchemaFromClasses(const TArray<UClass*>& Classes, const FString& CombinedSchemaPath)
+bool CheckVariableNameValidity(TSharedPtr<FUnrealType> TypeInfo)
+{
+	// Check Replicated Data
+	FUnrealFlatRepData RepData = GetFlatRepData(TypeInfo);
+
+	for (EReplicatedPropertyGroup Group : GetAllReplicatedPropertyGroups())
+	{
+		TMap<FString, TSharedPtr<FUnrealProperty>> SchemaReplicatedDataNames;
```

Comment:
Bear in mind that looking up `FString` is case-insensitive, so e.g. `Friend!!!Ship` and `Friendship?!?!` will collide.
I still think it's a valid restriction since Unreal itself doesn't allow identifiers that are the same except for case, but we should be explicit about it in the docs.